test_unstructured/partition/test_csv.py

```diff
@@ -263,15 +253,158 @@ def test_partition_csv_respects_languages_arg():
+                metadata_file_path=None,
+                metadata_last_modified=None,
+                include_header=True,
+                infer_table_structure=True,
+                date_from_file_object=False,
+            )
+
+    # -- .delimiter ---------------------------------------------
+
+    @pytest.mark.parametrize(
+        "file_name",
+        [
+            "stanley-cups.csv",
+            # -- Issue #2643: previously raised `_csv.Error: Could not determine delimiter` on
+            # -- this file
+            "csv-with-long-lines.csv",
+        ],
+    )
+    def it_auto_detects_the_delimiter_for_a_comma_delimited_CSV_file(self, file_name: str):
+        ctx = _CsvPartitioningContext(example_doc_path(file_name))
+        assert ctx.delimiter == ","
+
+    def and_it_auto_detects_the_delimiter_for_a_semicolon_delimited_CSV_file(self):
+        ctx = _CsvPartitioningContext(example_doc_path("semicolon-delimited.csv"))
+        assert ctx.delimiter == ";"
+
+    def but_it_returns_None_as_the_delimiter_for_a_single_column_CSV_file(self):
+        ctx = _CsvPartitioningContext(example_doc_path("single-column.csv"))
+        assert ctx.delimiter is None
+
+    # -- .header ------------------------------------------------
+
+    @pytest.mark.parametrize(("include_header", "expected_value"), [(False, None), (True, 0)])
+    def it_identifies_the_header_row_based_on_include_header_arg(
+        self, include_header: bool, expected_value: int | None
+    ):
+        assert _CsvPartitioningContext(include_header=include_header).header == expected_value
+
+    # -- .last_modified --------------------------
+
+    def it_gets_the_last_modified_date_of_the_document_from_the_caller_when_provided(self):
+        ctx = _CsvPartitioningContext(metadata_last_modified="2024-08-04T13:12:35")
+        assert ctx.last_modified == "2024-08-04T13:12:35"
+
+    def and_it_falls_back_to_the_last_modified_date_of_the_file_when_a_path_is_provided(
+        self, get_last_modified_date_: Mock
+    ):
+        get_last_modified_date_.return_value = "2024-08-04T02:23:53"
+        ctx = _CsvPartitioningContext(file_path="a/b/document.csv")
+
+        last_modified = ctx.last_modified
+
+        get_last_modified_date_.assert_called_once_with("a/b/document.csv")
+        assert last_modified == "2024-08-04T02:23:53"
+
+    def and_it_falls_back_to_last_modified_date_of_file_when_a_file_like_object_is_provided(
+        self, get_last_modified_date_from_file_: Mock
+    ):
+        get_last_modified_date_from_file_.return_value = "2024-08-04T13:17:47"
+        file = io.BytesIO(b"abcdefg")
+        ctx = _CsvPartitioningContext(file=file, date_from_file_object=True)
+
+        last_modified = ctx.last_modified
+
+        get_last_modified_date_from_file_.assert_called_once_with(file)
+        assert last_modified == "2024-08-04T13:17:47"
+
+    def but_it_falls_back_to_None_for_the_last_modified_date_when_date_from_file_object_is_False(
```

Comment:
Is there still a test case for "long lines" like the test that was removed:
```
def test_partition_csv_detects_the_right_csv_delimiter():
    assert get_delimiter("example-docs/csv-with-long-lines.csv") == ","
```

Reply:
Yeah that one moved into the unit tests as a test-case.